packages/pouch-storage/src/PouchStorageTable.ts

```diff
@@ -35,20 +35,17 @@ export type PouchDBSort = Array<
   string | { [propName: string]: 'asc' | 'desc' }
 >;
 
-// We may want to use `PouchDB.Find.ConditionOperators` instead of this, but
-// there are some mismatches in how we use this with the types.
-// https://github.com/deephaven/web-client-ui/issues/1505 to address this
 type PouchFilter = OnlyOneProp<{
   $eq: FilterValue | FilterValue[];
-  $neq: FilterValue | FilterValue[];
+  $ne: FilterValue | FilterValue[];
   $gt: FilterValue | FilterValue[];
   $gte: FilterValue | FilterValue[];
   $lt: FilterValue | FilterValue[];
   $lte: FilterValue | FilterValue[];
-  $regex: RegExp;
+  $regex: RegExp | string;
 }>;
```

Comment:
Here we should use the `ConditionOperators` types defined in `PouchDB` themselves instead of redefining them:

```
type PouchFilter = OnlyOneProp<
  Pick<PouchDB.Find.ConditionOperators, '$eq' | '$ne' | ...etc... >
>;

Note that the `RegExp` type is defined as a `string`, so we'll need to modify the switch-case below to `.toString()` the `RegExp`s (I think this is implicitly being done somewhere right now, we should be explicit with it though since this type doesn't match currently).

On a side note, the types for some of the operators like `$gt` are just `any`. Ideally those operators would have a more strict type, but we'll leave that up to the pouchdb-find package to enforce.
```

Reply:
@mofojed I was the one who encouraged this. The `ConditionOperators` type uses a number of `any` types such that we lose type safety.